include/ast.hpp

```diff
@@ -194,6 +194,69 @@ struct IntConstExprNode : public ExprNode {
   int val;
 };
 
+/// @note This is an abstract class.
+struct UnaryExprNode : public ExprNode {
+  UnaryExprNode(std::unique_ptr<ExprNode> clause) : clause{std::move(clause)} {}
+
+  virtual void Accept(NonModifyingVisitor&) const override;
+  virtual void Accept(ModifyingVisitor&) override;
+
+  /// @note To make the class abstract.
+  virtual ~UnaryExprNode() = 0;
+
+  std::unique_ptr<ExprNode> clause;
```

Comment:
I wonder about the rationale behind choosing the name `clause`, as I didn't find its use in the reference manual, specifically in section 6.5.3.